provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinBuildScriptCompiler.kt

```diff
@@ -476,16 +476,44 @@ inline fun <T> KotlinScriptSource.withLocationAwareExceptionHandling(action: ()
 
 
 private
-inline fun <T> LoadedScriptClass<T>.eval(action: LoadedScriptClass<T>.() -> Unit) =
+inline fun <T> LoadedScriptClass<T>.eval(scriptSource: KotlinScriptSource, action: LoadedScriptClass<T>.() -> Unit) =
     withContextClassLoader(scriptClass.classLoader) {
-        try {
-            action()
-        } catch (e: InvocationTargetException) {
-            throw e.targetException
-        }
+        withLocationAwareScriptEvaluationExceptionHandling(scriptSource, action)
     }
 
 
+private
+inline fun <T> LoadedScriptClass<T>.withLocationAwareScriptEvaluationExceptionHandling(
+    scriptSource: KotlinScriptSource,
+    action: LoadedScriptClass<T>.() -> Unit
+) =
+    try {
+        action()
+    } catch (e: Throwable) {
+        throw maybeLocateException(scriptSource, e)
```

Comment:
❓ Do you have issues with the `InterruptedException` here and needing to reset the thread's interrupt flag?

http://www.yegor256.com/2015/10/20/interrupted-exception.html

Reply:
It's not applicable here.

Reply:
I have no idea. That's why I'm asking.